2024-06-17/main.py

```diff
@@ -0,0 +1,201 @@
+import unittest
+import sqlite3
+import uuid
+from pprint import pprint
+from fastapi import FastAPI, APIRouter, HTTPException
+
+# Create our in-memory DB and cursor
+db = sqlite3.connect("file::memory:?cache=shared", check_same_thread=False)
+cur = db.cursor()
+
+# Create your table SQL statement
+def createDB():
+    cur.execute("CREATE TABLE users(id, name, address, email, phone)")
+
+
+# Create your table!
+createDB()
+
+# Instantiate our API/app
+app = FastAPI()
+router = APIRouter(prefix="/api")
+
+# Endpoint for adding a NEW user
+@router.post("/add/user")
+def add_user(name, address=None, email=None, phone=None) -> dict:
+    id = insertUser(name, address, email, phone)
+    return getUser(id)
+
+
+# Endpoint for getting ALL USERS
+@router.get("/get/all")
+def get_all_users(limit=10) -> list[dict]:
+    return getAllUsers(limit)
+
+
+# Endpoint for GETTING an EXISTING USER
+@router.get("/get/user")
+def get_user(id) -> dict:
+    return getUser(id)
+
+
+# Endpoint for UPDATING an EXISTING USER
+@router.patch("/update/user")
+def update_user(id, name=None, address=None, email=None, phone=None):
```

Comment:
Would you mind adding a return annotation, here?

Reply:
ues